### ISSUE TITLE
CustomComponent build method using default value of dict() crashes langflow editor (langflow alpha)

### ISSUE THREAD

**Issue by rothnic** [langflow-ai/langflow#2009]
**Describe the bug**
Flow editor crashes when custom component has a default value of `dict()` for one of the input options.

**Browser and Version**
 - Browser: Arc Version 1.43.1 (50198) Chromium Engine Version 125.0.6422.60
 - Langflow: v1.0.0a40 (self-hosted via docker)

**To Reproduce**
* I was playing around with input options on the build method of the custom component.
* I had something like `def build(options: dict = {})`
* I was intending to have an optional dict of values to be passed into the build call and the custom component didn't change so i tried: `def build(options: dict = dict()`. I often bounce between python and javascript, so wasn't sure of the correct way of doing this.
* I saved the component and then the flow editor crashed

**Screenshots**

<img width="871" alt="image" src="https://github.com/langflow-ai/langflow/assets/452052/4e8cfa63-5844-4665-a1ac-ba7aad617328">

<img width="751" alt="image" src="https://github.com/langflow-ai/langflow/assets/452052/94f7b61b-c0be-46a1-a844-047b1a594a92">

**Additional context**

Dev console output:

```SyntaxError: Unexpected token 'd', "dict()" is not valid JSON
    at JSON.parse (<anonymous>)
    at convertObjToArray (index-94993d38.js:3096:28971)
    at index-94993d38.js:3647:20396
    at Array.map (<anonymous>)
    at index-94993d38.js:3647:17606
    at Nh (index-94993d38.js:981:18734)
    at Yi (index-94993d38.js:983:1694)
    at Vk (index-94993d38.js:983:50181)
    at Uk (index-94993d38.js:983:43292)
    at Tk (index-94993d38.js:983:43214)```

**Comment by rothnic** [comment]
Worth noting that I can refresh the page to get back to the flows list, then i get the error each time i try to open the flow again. Probably want some way to recover a flow if an error like this happens in general

**Comment by anovazzi1** [comment]
Hello @rothnic,
Sorry for the delay. Did you try using the new version? Does the error still persist?

**Comment by carlosrcoelho** [comment]
Hi @rothnic ,
 
I hope you're doing well. Just a friendly reminder that if we do not hear back from you within the next 3 days, we will close this issue. If you need more time or further assistance, please let us know. 

Thank you for your understanding!

**Comment by rothnic** [comment]
> Hi @rothnic ,   I hope you're doing well. Just a friendly reminder that if we do not hear back from you within the next 3 days, we will close this issue. If you need more time or further assistance, please let us know. 
> 
> Thank you for your understanding!

Has this issue been addressed specifically or you just need me to create another component with a dict as demonstrated above to see if it is still an issue? I'm about to be on vacation for the next week and a half so likely can't test this in the near term.

It seems like robustness to various input options would be something that would be important to have unit testing for to induce odd situations, given the support for custom components. Is this something that has been added since I submitted this?
